tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,726 @@
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_PLAY_MEDIA,
+        {
+            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+            ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+        },
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.state == initial_state.state
+    assert state.last_updated > initial_state.last_updated
+
+
+async def test_async_play_media_from_stopped(
+    hass, get_request_return_values, mock_api_object
+):
+    """Test async play media from stopped."""
+    updater_update = mock_api_object.start_websocket_handler.call_args[0][2]
+
+    get_request_return_values["player"] = SAMPLE_PLAYER_STOPPED
+    await updater_update(["player"])
+    await hass.async_block_till_done()
+    initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_PLAY_MEDIA,
+        {
+            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+            ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+        },
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.state == initial_state.state
+    assert state.last_updated > initial_state.last_updated
+
+
+async def test_async_play_media_unsupported(hass, mock_api_object):
+    """Test async play media on unsupported media type."""
+    initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_PLAY_MEDIA,
+        {
+            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_TVSHOW,
+            ATTR_MEDIA_CONTENT_ID: "wontwork.mp4",
+        },
+    )
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.last_updated == initial_state.last_updated
+
+
+async def test_async_play_media_tts_timeout(hass, mock_api_object):
+    """Test async play media with TTS timeout."""
+    mock_api_object.add_to_queue.side_effect = None
+    with patch("homeassistant.components.forked_daapd.media_player.TTS_TIMEOUT", 0):
+        initial_state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+        await _service_call(
+            hass,
+            TEST_MASTER_ENTITY_NAME,
+            SERVICE_PLAY_MEDIA,
+            {
+                ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+                ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+            },
+        )
+        state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+        assert state.state == initial_state.state
+        assert state.last_updated > initial_state.last_updated
+
+
+async def test_use_pipe_control_with_no_api(hass, mock_api_object):
+    """Test using pipe control with no api set."""
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_SELECT_SOURCE,
+        {ATTR_INPUT_SOURCE: "librespot-java (pipe)"},
+    )
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PLAY)
+    assert mock_api_object.start_playback.call_count == 0
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)